### PR TITLE
feat: potential fix for premiumfix

### PR DESCRIPTION
**Pull request by DEV-DIBSTER**
@danielpmc @FC5570 Review.

### CODE REVIEW

File: bot/discord/commands/staff/premiumfix.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ exports.run = async(client, message, args) => {
     } else {
         const replyMsg = await message.channel.send('Staring calculation...');
 
-        let selectedUser = message.mentions.users.first() || message.guild.members.fetch(args[1]);
+        let selectedUser = message.mentions.users.first() || message.guild.members.fetch(args[1]); || message.client.users.cache.get(args[1].match(/[0-9]{18}/).length == 0 ? args[1] : args[1].match(/[0-9]{18}/)[0]);
```
Comment (WilliamDavidHarrison) [inline bot/discord/commands/staff/premiumfix.js:12]: > ```js
> args[1].match(/[0-9]{18}/)[0])
> ```

This should be-
```js
args[1].match(/[0-9]{19}/)[0])
```
-to match the new user IDs.

**Review by WilliamDavidHarrison — CHANGES_REQUESTED** [review]

**Comment by Sliden101** [comment]
that regex checks for 18 and up if it's 19 old ids won't work

File: bot/discord/commands/staff/premiumfix.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ exports.run = async(client, message, args) => {
     } else {
         const replyMsg = await message.channel.send('Staring calculation...');
 
-        let selectedUser = message.mentions.users.first() || message.guild.members.fetch(args[1]);
+        let selectedUser = message.mentions.users.first() || message.guild.members.fetch(args[1]); || message.client.users.cache.get(args[1].match(/[0-9]{19}/).length == 0 ? args[1] : args[1].match(/[0-9]{19}/)[0]);
```
Comment (itzJOHv) [inline bot/discord/commands/staff/premiumfix.js:12]:
```suggestion
        let selectedUser = message.mentions.users.first() || message.guild.members.fetch(args[1]); || message.client.users.cache.get(args[1].match(/[0-9]{18,19}/).length == 0 ? args[1] : args[1].match(/[0-9]{18,19}/)[0]);
```
:Okay:

**Review by itzJOHv — CHANGES_REQUESTED** [review]

File: bot/discord/commands/staff/premiumfix.js (hunk 2 above)
Comment (WilliamDavidHarrison) [reply bot/discord/commands/staff/premiumfix.js:12]: Some users have 17 digit IDs

Comment (itzJOHv) [reply bot/discord/commands/staff/premiumfix.js:12]: > Some users have 17 digit IDs

edited thanks

**Pull request merged**